src/plugins/intel_gpu/src/runtime/memory.cpp

```diff
@@ -45,4 +45,107 @@ std::unique_ptr<surfaces_lock> surfaces_lock::create(engine_types engine_type, s
     }
 }
 
+ov::Shape MemoryUsageTracker::shapes_math(const ov::Shape& shape1, const ov::Shape& shape2, math_op op) {
```

Comment:
I think several operators(=,-,*) for ov::Shape would look better. I.e.

```cpp
namespace ov {
static ov::Shape operator+(const ov::Shape& s1, const ov::Shape& s2) { }
static ov::Shape operator-(const ov::Shape& s1, const ov::Shape& s2) { }
static ov::Shape operator*(const ov::Shape& s1, const ov::Shape& s2) { }
} // namespace ov

...

        for (size_t i = 0; i < min_shapes_num - 1; ++i) {
            auto result = shapes[shapes_num - i - 1] - shapes[shapes_num - i - 2];
```

Reply:
Good idea, thanks